### PR TITLE
first version of injector

### PR DESCRIPTION
**Pull request by oleksiistepaniak**

### CODE REVIEW

File: src/main/java/mate.academy/lib/Injector.java [hunk 1]
```diff
@@ -1,13 +1,72 @@
...
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed, missing @Component annotation"
+                    + "on the class " + interfaceClazz.getName());
+        }
+        Object clazzImplementationInstance = null;
```
Comment (Andriy-dz) [inline src/main/java/mate.academy/lib/Injector.java]: I'd recommend checking if class implementation is present in instances map first. And return the already present instance immediately, if present

File: src/main/java/mate.academy/lib/Component.java [hunk 2]
```diff
@@ -1,5 +1,9 @@
 package mate.academy.lib;
 
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
```
[the remark this answers is not in the document]
Comment (Andriy-dz) [reply src/main/java/mate.academy/lib/Component.java:7]: agree with @AndrewkeY-hub

File: src/main/java/mate.academy/lib/Inject.java [hunk 3]
```diff
@@ -1,5 +1,9 @@
 package mate.academy.lib;
 
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
```
[the remark this answers is not in the document]
Comment (Andriy-dz) [reply src/main/java/mate.academy/lib/Inject.java:7]: agree with @AndrewkeY-hub

File: src/main/java/mate.academy/lib/Injector.java [hunk 4]
```diff
@@ -1,13 +1,72 @@
...
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value."
+                            + "Class: " + clazz.getName() + ". Field: " + field.getName());
```
Comment (Andriy-dz) [inline src/main/java/mate.academy/lib/Injector.java:42]: let's add the error itself

**Review by Andriy-dz — CHANGES_REQUESTED** [review]
Good Job! Let`s impove your solution

**Review by AndrewkeY-hub — APPROVED** [review]

**Review by Andriy-dz — APPROVED** [review]
Perfect!

**Pull request closed without merging**